examples/flutter_gallery/lib/gallery/backdrop.dart

```diff
@@ -0,0 +1,329 @@
+
+    final double opacity1 = new CurvedAnimation(
+      parent: new ReverseAnimation(progress),
+      curve: const Interval(0.5, 1.0),
+    ).value;
+
+    final double opacity2 = new CurvedAnimation(
+      parent: progress,
+      curve: const Interval(0.5, 1.0),
+    ).value;
+
+    return new Stack(
+      alignment: alignment,
+      children: <Widget>[
+        new IgnorePointer(
+          ignoring: opacity1 < 1.0,
+          child: new Opacity(
+            opacity: opacity1,
+            child: child1,
+          ),
+        ),
+        new IgnorePointer(
+          ignoring: opacity2 <1.0,
+          child: new Opacity(
+            opacity: opacity2,
+            child: child0,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _BackAppBar extends StatelessWidget {
+  const _BackAppBar({
+    Key key,
+    this.leading: const SizedBox(width: 56.0),
+    @required this.title,
+    this.trailing,
+  }) : assert(leading != null), assert(title != null), super(key: key);
+
+  final Widget leading;
+  final Widget title;
+  final Widget trailing;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[
+      new Container(
+        alignment: Alignment.center,
+        width: 56.0,
+        child: leading,
+      ),
+      new Expanded(
+        child: title,
+      ),
+    ];
+
+    if (trailing != null) {
+      children.add(
+        new Container(
+          alignment: Alignment.center,
+          width: 56.0,
+          child: trailing,
+        ),
+      );
+    }
+
+    final ThemeData theme = Theme.of(context);
+
+    return IconTheme.merge(
+      data: theme.primaryIconTheme,
+      child: new DefaultTextStyle(
+        style: theme.primaryTextTheme.title,
+        child: new SizedBox(
+          height: _kBackAppBarHeight,
+          child: new Row(children: children),
+        ),
+      ),
+    );
+  }
+}
+
+class Backdrop extends StatefulWidget {
+  const Backdrop({
+    this.frontAction,
+    this.frontTitle,
+    this.frontHeading,
+    this.frontLayer,
+    this.backTitle,
+    this.backLayer,
+  });
+
+  final Widget frontAction;
+  final Widget frontTitle;
+  final Widget frontLayer;
+  final Widget frontHeading;
+  final Widget backTitle;
+  final Widget backLayer;
+
+  @override
+  _BackdropState createState() => new _BackdropState();
+}
+
+class _BackdropState extends State<Backdrop> with SingleTickerProviderStateMixin {
+  final GlobalKey _backdropKey = new GlobalKey(debugLabel: 'Backdrop');
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = new AnimationController(
+      duration: const Duration(milliseconds: 300),
+      value: 1.0,
+      vsync: this,
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  double get _backdropHeight {
+    final RenderBox renderBox = _backdropKey.currentContext.findRenderObject();
```

Comment:
I think it might throw a cryptic error if future maintainers called this at the wrong time. Can we pre-assert something here?

Reply:
I've added a comment about not calling the getter at build time (which the code does not do). If someone does change things and call this getter at the wrong time, they'll find the comment but not satisfaction :-).